### DISCUSSION TITLE
Pytorch 1.8 downgraded

### DISCUSSION THREAD

**Discussion by rbunn80110** [KeremTurgutlu/self_supervised#52] [Q&A] [answered]
When I install self-supervised this happens:

Uninstalling torch-1.8.0+cu111

Does pytorch 1.8 not work with the code base?

**Comment by KeremTurgutlu** [comment, marked as answer]
Since fastai currently depends on `pip_requirements = torch>=1.7.0,<1.8` it will be downgraded during installation. Once fastai moves to >1.8, it will support it.

**Reply by rbunn80110** [reply]
Actually, when I install pip fastai it isn't downgraded.  When I install self-supervised is when the downgrade happens

**Reply by rbunn80110** [reply]
If pytorch is already installed on the system fastai doesn't downgrade it

**Reply by rbunn80110** [reply]
Attempting uninstall: fastai
    Found existing installation: fastai 1.0.61
    Uninstalling fastai-1.0.61:
      Successfully uninstalled fastai-1.0.61
Successfully installed catalogue-1.0.0 fastai-2.3.0 fastcore-1.3.19 kornia-0.5.0 plac-1.1.3 self-supervised-1.0.2 spacy-2.3.5 srsly-1.0.5 thinc-7.4.5 timm-0.4.5 torch-1.7.1 torchvision-0.8.2

**Reply by rbunn80110** [reply]
Maybe the previous version of fastai let newer versions of pytorch slide

**Reply by rbunn80110** [reply]
It's not a big deal though.  Just bringing it to your attention

**Reply by KeremTurgutlu** [reply]
Recommended fastai installation is `pip install -U fastai`
